### PR TITLE
Add interpreter

### PR DESCRIPTION
**Pull request by enotvtapke**

### CODE REVIEW

File: src/test/kotlin/InterpreterTest.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import Scoped.Companion.scoped
+import ast.Statement
+import interpreter.Interpreter
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class InterpreterTest {
```
Comment (enotvtapke) [inline src/test/kotlin/InterpreterTest.kt:7]: Please add test for interpretation of negative numbers

File: src/main/kotlin/lexer/Terminal.kt [hunk 2]
```diff
@@ -7,12 +7,12 @@ enum class Terminal(val pattern: Regex? = null) {
     LBRACKET("\\{".toRegex()),
     RBRACKET("}".toRegex()),
     VAR("[a-z][A-Za-z_0-9]*".toRegex()),
-    INT("[0-9]+".toRegex()),
+    INT("-?[0-9]+".toRegex()),
```
Comment (enotvtapke) [inline src/main/kotlin/lexer/Terminal.kt:10]: Do we really need support for negative numbers?

File: src/main/kotlin/lexer/Lexer.kt [hunk 3]
```diff
@@ -38,7 +38,7 @@ open class Lexer(private val input: CharSequence) {
         while (curTerminal.terminal == WS) nextToken()
     }
 
-    private fun currentPrefix(length: Int = 15): CharSequence =
+    fun currentPrefix(length: Int = 15): CharSequence =
```
Comment (enotvtapke) [inline src/main/kotlin/lexer/Lexer.kt:41]: Why default value is 15?

File: src/main/kotlin/interpreter/EvalEnvironment.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package interpreter
+
+class EvalEnvironment {
```
Comment (enotvtapke) [inline src/main/kotlin/interpreter/EvalEnvironment.kt:3]: I don't really like this name. Rename it to InterpreterEnvironment

File: src/main/kotlin/lexer/Lexer.kt (hunk 3 above)
Comment (enotvtapke) [inline src/main/kotlin/lexer/Lexer.kt:41]: Why default value is 15?

File: src/test/kotlin/InterpreterTest.kt (hunk 1 above)
Comment (enotvtapke) [inline src/test/kotlin/InterpreterTest.kt:7]: Please add test for interpretation of negative numbers

File: src/main/kotlin/parser/Parser.kt [hunk 5]
```diff
@@ -78,7 +78,13 @@ class Parser(input: CharSequence) {
 
     fun statements(): List<Statement> {
```
Comment (enotvtapke) [inline src/main/kotlin/parser/Parser.kt:79]: Verify that this function is syntactically correct

Comment (enotvtapke) [inline src/main/kotlin/parser/Parser.kt:79]: erify that this function is syntactically correct
